Fix template values when using USE_THOUSAND_SEPARATOR=True

SebCorbin opened the pull request:
In Wagtail `6.2.1`, when using `USE_THOUSAND_SEPARATOR=True` in settings, a large revision ID would be sent localized through ajax, resulting in a `ValueError` in `wagtail.admin.views.editing_sessions.ping`:

```
ValueError: invalid literal for int() with base 10: '4\xa0112'
  File "django/db/models/fields/__init__.py", line 2053, in get_prep_value
    return int(value)

ValueError: Field 'id' expected a number but got '4\xa0112'.
  File "django/core/handlers/exception.py", line 55, in inner
    response = get_response(request)
  File "django/core/handlers/base.py", line 185, in _get_response
    response = middleware_method(
  File "rgive/back/middleware.py", line 39, in process_view
    return login_required(view_func, redirect_field_name=None)(
  File "django/contrib/auth/decorators.py", line 23, in _wrapper_view
    return view_func(request, *args, **kwargs)
  File "django/views/decorators/cache.py", line 62, in _wrapper_view_func
    response = view_func(request, *args, **kwargs)
  File "wagtail/admin/urls/__init__.py", line 172, in wrapper
    return view_func(request, *args, **kwargs)
  File "wagtail/admin/auth.py", line 151, in decorated_view
    response = view_func(request, *args, **kwargs)
  File "django/views/decorators/http.py", line 43, in inner
    return func(request, *args, **kwargs)
  File "wagtail/admin/views/editing_sessions.py", line 101, in ping
    original_revision = all_revisions.get(id=revision_id)
  ...
```

`\xa0` being a JSON-encoded comma

I took the liberty to fix other template that could have this problem:
 - the `parent_id` in select_all checkbox of a bulk action
 - the `image_id` in a image chooser modal results

laymonage commented on wagtail/admin/templates/wagtailadmin/shared/editing_sessions/module.html:
```diff
@@ -36,7 +36,7 @@
     {% if revision_id %}
         {# The ID of the latest revision that was known when this module was initially loaded #}
-        <input type="hidden" name="revision_id" value="{{ revision_id }}" />
+        <input type="hidden" name="revision_id" value="{{ revision_id|unlocalize }}" />
```
I think we're better off casting the ID to string in the Python code like we did for the interval:

https://github.com/wagtail/wagtail/blob/a59a2d0585f9b2ee4ff94152a5de4c66b02819df/wagtail/admin/ui/editing_sessions.py#L33-L35

The `|unlocalize` in the template for the ping interval was mistakenly added during merging because we thought we haven't casted it to string. We should remove that.

Also, could you add a test for this? I think you should be able to add another method that calls an existing test method, but with `override_settings` in https://github.com/wagtail/wagtail/blob/a59a2d0585f9b2ee4ff94152a5de4c66b02819df/wagtail/admin/tests/test_editing_sessions.py#L1336 e.g.

```py
@override_settings(USE_THOUSAND_SEPARATOR=True)
def test_edit_view_with_thousand_separator(self):
    # assertRevisionInput will fail if the revision_id is missing |unlocalize
	self.test_edit_view_with_default_interval()
```

Since this has bitten us a few times before – I'm starting to think we really should update our test matrix to have at least one with `USE_THOUSAND_SEPARATOR = True`...

SebCorbin replied:
Test added. As for

> Alternatively, we might be able to run the entire test suite with USE_THOUSAND_SEPARATOR = True and see if there are already failing tests for them.

The thing is we must set a context a large numbers every time so I don't think adding a setting in the matrix would be enough, extra care for template variable usage from reviewers should go a long way.

laymonage reviewed (commented):
Thanks! Could you add tests for this? I've added some instructions on how to add one for the editing session. I can't give you detailed instructions for the others at the moment, but let me know if you need any help.

Alternatively, we might be able to run the entire test suite with `USE_THOUSAND_SEPARATOR = True` and see if there are already failing tests for them.

SebCorbin commented:
Okay so after adding this test for select all, there was no fail, so I assume the value is cast to string somewhere, I removed the test because the `load_bulk()` would take 3 seconds which seems like a lifetime in unit tests.

```python
    def test_index_with_thousand_separator(self):
        Collection.load_bulk(
            [{'data': {'name': f'Collection {i}'}} for i in range(1_000)])
        root_collection = Collection.get_first_root_node()
        travel_plans = root_collection.add_child(name="Travel plans")
        self.assertGreater(travel_plans.pk, 1000)
        models.Document.objects.create(title="Itinerary", collection=travel_plans)

        response = self.get({"collection_id": travel_plans.pk})
        # Should render the "Select all" with the correct collection ID
        # twice, one for the column header and one in the footer.
        self.assertContains(
            response,
            f"""
            <input data-bulk-action-parent-id="{travel_plans.pk}"
                   data-bulk-action-select-all-checkbox
                   type="checkbox"
                   aria-label="Select all"
            />""",
            html=True,
            count=2,
        )
```

As for `iamge.id`, I'm looking into reproducing it in tests right now.

SebCorbin commented:
Forget my previous comment, I was missing `override_settings(USE_THOUSAND_SEPARATOR=True)`, so I added back the code. I also added the test for the image chooser too.

gasman commented:
Fixed for stable/6.2.x in 4604ec8a5bf2ef9561307c2250d2c6e7e0bab9f0, and for main in #12251.
